### PR TITLE
[ADD] payment_providers/worldline

### PR DESCRIPTION
**Pull request by Demesmaeker**
New payment provider, it's features, and how to set it up

task-3499275

See also:
- https://github.com/odoo/odoo/pull/175194

### CODE REVIEW

File: content/applications/finance/payment_providers/worldline.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+You need to create an **API user** to be used in the creation of transactions from Odoo. While you
+can use your main account to do so, using an **API user** ensures that if the credentials used in
+Odoo are leaked, no access to your Worldline configuration is possible. Additionally, passwords for
+**API users** do not need to be updated regularly, unlike normal users.
+
+To create an **API user**, go to :menuselection:`Configuration --> Users` and click on
+:guilabel:`New User`. The following fields must be configured:
+
+.. _worldline/worldline:
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/worldline.rst:29]: I would change the name of this anchor to something more specific, e.g.,:
```suggestion
.. _worldline/API-user:
```

File: content/applications/finance/payment_providers/worldline.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+- :guilabel:`UserID`: you can choose anything you want.
+- :guilabel:`User's Name, E-mail and Timezone`: you can enter the information you want.
+- :guilabel:`Profile`: should be set to :guilabel:`Admin`.
+- :guilabel:`Special user for API`: should be checked.
+
+.. tip::
+   If you already have an user set up, make sure it is activated without any error.
+
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/worldline.rst:38]: I would add an anchor here so we can use it in the Odoo section

File: content/applications/finance/payment_providers/worldline.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+From your merchant portal, head to :menuselection:`Developer --> Payment API`, click on
+:guilabel:`Add API key`
+
+Write down your :guilabel:`API key ID` and the associated :guilabel:`Secret API key`, you'll need
+these to set up your provider.
+
+When done, head to :menuselection:`Developer --> Webhooks` and :guilabel:`generate webhooks keys`.
+
+Write down your :guilabel:`Webhook key` and the associated :guilabel:`Webhook secret`, you'll need
+these to set up your provider.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/worldline.rst:53]:
```suggestion
#. Go to :menuselection:`Developer --> Webhooks` and click on :guilabel:`Generate webhook keys`. Copy the :guilabel:`Webhook ID` and the associated :guilabel:`Secret webhook key` and save them for later.
```

File: content/applications/finance/payment_providers/worldline.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+On the same page, you can now :guilabel:`Add webhook endpoint`.
+
+There, you can enter the **URL**, it should contain the same following URL, with `<example>`
+replaced by your database: `https://<example>/payment/worldline/webhook`.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/worldline.rst:58]: Is this absolutely necessary? If so I would rephrase to make that clear (to me "you can" could be interpreted as "it's optional"):
```suggestion
#. Click :guilabel:`Add webhook endpoint`, enter your Odoo database's URL followed by `/payment/worldline/webhook` in the :guilabel:`Endpoint url` field, and :guilabel:`Confirm`. For example: `https://example.odoo.com/payment/worldline/webhook`.

```

File: content/applications/finance/payment_providers/worldline.rst [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+
+Settings in Odoo
+================
+
+To set up Worldline in Odoo, head to :menuselection:`Accounting --> Configuration --> Payment
+Providers` and open the Worldline provider. In the :guilabel:`Credentials` tab, enter the **PSPID**
+of your Worldline account, and fill out the other fields as configured in your :ref:`Worldline
+portal <worldline/worldline>`.
```
Comment (auva-odoo) [inline content/applications/finance/payment_providers/worldline.rst:67]:
```suggestion
To set up Worldline in Odoo:
#. :ref:`Navigate to the payment provider Worldline <payment_providers/add_new>` and change its state to :guilabel:`Enabled`. 
#. In the :guilabel:`Credentials` tab, enter the :guilabel:`PSPID` of your Worldline account and fill in the :guilabel:`API Key`, :guilabel:`API Secret`, :guilabel:`Webhook Key`, and :guilabel:`Webhook Secret` with the values you saved at the step :ref:`Set up Worldline for Odoo <add the new anchor here>`. 
#. Configure the rest of the options to your liking.
```

**Review by auva-odoo — CHANGES_REQUESTED** [review]
Hi @Demesmaeker thank you for your work on this! I've added some comments and suggestions to help clarify the documentation and align it with other payment provider docs like Razorpay and APS. 
Also, correct me if I'm wrong, but should we retarget this PR to 18? That might also resolve the ci/minimal check issue I think? Don't hesitate if you have any questions (or if you'd prefer I do the changes myself). Thanks again!

File: content/applications/finance/payment_providers/worldline.rst [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+Worldline
+=========
+
+`Worldline <https://worldline.com/>`_, is a France-based company and the world's number 4 payments
+leader.
+
+.. seealso::
+   - :ref:`payment_providers/add_new`
+   - `Worldline's documentation <https://worldline.com/en/home/main-navigation/solutions/merchants/online-payment-gateway>`_.
+
```
[the remark this answers is not in the document]
Comment (Demesmaeker) [reply content/applications/finance/payment_providers/worldline.rst:11]: I don't know, there might be one, but not sure what we should put here instead '^^

File: content/applications/finance/payment_providers/worldline.rst [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+Log into your Worldline Merchant Portal and from the menu, go to the :guilabel:`Back Office`. From
+there, head to the :guilabel:`Configuration` tab.
+
+You need to create an **API user** to be used in the creation of transactions from Odoo. While you
+can use your main account to do so, using an **API user** ensures that if the credentials used in
+Odoo are leaked, no access to your Worldline configuration is possible. Additionally, passwords for
+**API users** do not need to be updated regularly, unlike normal users.
+
+To create an **API user**, go to :menuselection:`Configuration --> Users` and click on
+:guilabel:`New User`. The following fields must be configured:
```
[the remark this answers is not in the document]
Comment (Demesmaeker) [reply content/applications/finance/payment_providers/worldline.rst]: I used your suggestion with a general link =]

**Review by Demesmaeker — COMMENTED** [review]
Hello @auva-odoo 
I rebased. I opened that PR before the freeze, knowing it would happen before we could merge, thus the branch name and wrong target. ^^

I push the changes I could, feel free to change and push after me, I only did the basis so that nothing would be openly wrong according to the code, but I might not be the clearest documentation wise '^^

File: content/applications/finance/payment_providers/worldline.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
+=========
+Worldline
+=========
+
+`Worldline <https://worldline.com/>`_, is a France-based company and the world's number 4 payments
```
Comment (AntoineVDV) [inline content/applications/finance/payment_providers/worldline.rst:5]:
```suggestion
`Worldline <https://worldline.com/>`_, is a France-based company and the world's number four payments
```

File: content/applications/finance/payment_providers/worldline.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+not require frequent password updates like regular accounts.
+
+To create an **API user**,
+
+#. Log into your `Worldline Merchant Portal <https://merchant-portal.preprod.worldline-solutions.com/dashboard>`_,
+   click the :icon:`fa-th` (:guilabel:`menu`) icon and select :guilabel:`Back Office`.
+#. Go to :menuselection:`Configuration --> Users` and click on :guilabel:`New User`.
+#. Configure the following fields:
+
+   - Specify a :guilabel:`UserID, `User's name, E-mail address`, and :guilabel:`Timezone` of your
```
Comment (AntoineVDV) [inline content/applications/finance/payment_providers/worldline.rst:34]: Missing guilabels.

File: content/applications/finance/payment_providers/worldline.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+     choice.
+   - Set the :guilabel:`Profile` field to :guilabel:`Admin`.
+   - Enable :guilabel:`Special user for API`.
```
Comment (AntoineVDV) [inline content/applications/finance/payment_providers/worldline.rst:37]: Make it a numbered list too.

**Review by AntoineVDV — APPROVED** [review]

File: content/applications/finance/payment_providers/worldline.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (auva-odoo) [reply content/applications/finance/payment_providers/worldline.rst:11]: @AntoineVDV any ideas here?

[the remark this answers is not in the document]
Comment (AntoineVDV) [reply content/applications/finance/payment_providers/worldline.rst:11]: I don't see the point in including this link; users who read the current documentation will already have a Worldline account.

**Review by auva-odoo — APPROVED** [review]
@robodoo r+
